homeassistant/components/roborock/switch.py

```diff
@@ -0,0 +1,110 @@
+from .device import RoborockCoordinatedEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class RoborockSwitchDescriptionMixin:
+    """Define an entity description mixin for switch entities."""
+
+    # Gets the status of the switch
+    get_value: Callable[[RoborockCoordinatedEntity], Coroutine[Any, Any, dict]]
+    # Evaluate the result of get_value to determine a bool
+    evaluate_value: Callable[[dict], bool]
+    # Sets the status of the switch
+    set_command: Callable[[RoborockCoordinatedEntity, bool], Coroutine[Any, Any, dict]]
+
+
+@dataclass
+class RoborockSwitchDescription(
+    SwitchEntityDescription, RoborockSwitchDescriptionMixin
+):
+    """Class to describe an Roborock switch entity."""
+
+
+SWITCH_DESCRIPTIONS: list[RoborockSwitchDescription] = [
+    RoborockSwitchDescription(
+        set_command=lambda entity, value: entity.send(
+            RoborockCommand.SET_CHILD_LOCK_STATUS, {"lock_status": 1 if value else 0}
+        ),
+        get_value=lambda data: data.send(RoborockCommand.GET_CHILD_LOCK_STATUS),
+        evaluate_value=lambda data: data["lock_status"] == 1,
+        key="child_lock",
+        translation_key="child_lock",
+        icon="mdi:account-lock",
+        entity_category=EntityCategory.CONFIG,
+    )
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Roborock switch platform."""
+
+    coordinators: dict[str, RoborockDataUpdateCoordinator] = hass.data[DOMAIN][
+        config_entry.entry_id
+    ]
+    async_add_entities(
+        (
+            RoborockSwitchEntity(
+                f"{description.key}_{slugify(device_id)}",
+                coordinator,
+                description,
+            )
+            for device_id, coordinator in coordinators.items()
+            for description in SWITCH_DESCRIPTIONS
+        ),
+        True,
+    )
+
+
+class RoborockSwitchEntity(RoborockCoordinatedEntity, SwitchEntity):
+    """A class to let you turn functionality on Roborock devices on and off."""
+
+    entity_description: RoborockSwitchDescription
+
+    def __init__(
+        self,
+        unique_id: str,
+        coordinator: RoborockDataUpdateCoordinator,
+        entity_description: RoborockSwitchDescription,
+    ) -> None:
+        """Create a switch entity."""
+        self.entity_description = entity_description
+        super().__init__(unique_id, coordinator)
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        await self.entity_description.set_command(self, False)
+        return self.async_schedule_update_ha_state(True)
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn on the switch."""
+        await self.entity_description.set_command(self, True)
+        return self.async_schedule_update_ha_state(True)
+
+    async def async_update(self) -> None:
```

Comment:
Why are we overwriting this method of the coordinator entity? This will block users from manually updating the coordinator.

Reply:
I saw that this was a seemingly common thing to do when you need to update a switch async. since is_on is a non async property.

I got this approach from a wide variety of other integrations

Reply:
It's common for polling entities that don't use the coordinator but the built-in entity update mechanism.

This switch state doesn't seem connected to the coordinator data. Is it? Should we even inherit the coordinator entity here?

Reply:
It is connected to the coordinator as the coordinator holds the api. 

I guess a better solution is for me to get the value on a coordinator update and store it inside the coordinator?

Reply:
It depends on how the API returns data. If it returns multiple data points, you want to store it inside a coordinator so you can share it among entities.

Reply:
You don’t want to pass the coordinator around if you only care about `api`.

Reply:
The coordinator is meant to fetch data for more than a single data point, eg one data point being one entity state. If the entity needs to fetch entity specific state using the coordinator is probably not the correct approach for fetching the data.

I'd add a specific entity method that updates the entity state. We can still inherit CoordinatorEntity if we need the auxiliary coordinator state for device info eg.

Reply:
Ie, we can keep using `async_update` here but then we should also call `super().async_update` to also request to update the coordinator if the entity relies on both coordinator state and entity specific state.

Reply:
So far the switches I have only rely on entity specific state, so I'll throw together a Non-coordinated entity and use that - a lot of things can be determined from the status/props that the coordinator holds, for those that can't, I'll use a non coordinated entity and directly send the command. I don't think I'll have a PR for that tonight, but I should tomorrow morning.

If in the future any rely on both coordinator state and entity state, I'll create a separate coordinated switch entity 

Sorry for this misunderstanding! Thank you for the insight!

Reply:
Do I need to get this completed before the beta release tomorrow? If so - should I revert this for now? Or just commit the temporary solution of adding a super() call to async update?

Reply:
I think adding a call to `super().async_update` is ok for now.

Reply:
Done here #93840

Also added a extra switch, but can remove if it makes things easier

Reply:
Please address my comments in a separate PR that I can merge immediately. I don't prioritize further changes, sorry.

Reply:
Understood. I finished the non coordinated entity faster than I thought I would, are you okay with my making that one a PR or would you prefer for me to just keep it simple and leave it on the super().async_update() for now?

Reply:
I can take a look at that.

Reply:
#93841

Thank you again for your help!